robottelo/host_info.py

```diff
@@ -99,28 +99,52 @@ def get_repo_files(repo_path, extension='rpm', hostname=None):
     return sorted(repo_file for repo_file in result.stdout if repo_file)
 
 
-def get_repomd_revision(repo_path, hostname=None):
-    """Fetches a revision of repository.
+def get_repo_files_by_url(url, extension='rpm'):
+    """Returns a list of repo files (for example rpms) in a specific repository
+    published at some url.
 
-    :param str repo_path: unix path to the repo, e.g. '/var/lib/pulp/fooRepo'
-    :param str optional hostname: hostname or IP address of the remote host. If
-        ``None`` the hostname will be get from ``main.server.hostname`` config.
+    :param url: url where the repo or CV is published
+    :param extension: extension of searched files. Defaults to 'rpm'
+    :return:  list representing rpm package names
+    """
+    if not url.endswith('/'):
+        url += '/'
+    result = ssh.command(
+        f"""curl -k {url} | grep -oP '(?<=href=").*?(?=">)'""",
+    )
+    if result.return_code != 0:
+        raise CLIReturnCodeError(result.return_code, result.stderr, f'Failed to curl {url}')
+
+    if 'Packages/' not in result.stdout:
+        return sorted(line for line in result.stdout if extension in line)
+
+    files = []
+    subs = get_repo_files_by_url(f'{url}Packages/', extension='/')
+    for sub in subs:
+        files.extend(get_repo_files_by_url(f'{url}Packages/{sub}', extension))
+
+    return sorted(files)
+
+
+def get_repomd_revision(repo_url):
+    """Fetches a revision of a repository.
+
+    :param str repo_url: the 'Published_At' link of a repo
     :return: string containing repository revision
     :rtype: str
     """
     repomd_path = 'repodata/repomd.xml'
     result = ssh.command(
-        f"grep -oP '(?<=<revision>).*?(?=</revision>)' {repo_path}/{repomd_path}",
-        hostname=hostname,
+        f"curl -k {repo_url}/{repomd_path} | grep -oP '(?<=<revision>).*?(?=</revision>)'",
```

Comment:
This would be a good opportunity to make this request directly, instead of using ssh to run curl from the target satellite.

Reply:
same palmface here